spock-core/src/main/java/spock/lang/Tag.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package spock.lang;
+
+import org.spockframework.runtime.extension.ExtensionAnnotation;
+import org.spockframework.runtime.extension.builtin.TagExtension;
+import org.spockframework.util.Beta;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Indicates that a feature method or specification belongs to a certain user-defined category.
+ *
+ * @since 2.2
+ * @author Alexander Kriegisch
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
+@ExtensionAnnotation(TagExtension.class)
+@Repeatable(Tag.Container.class)
+public @interface Tag {
+  /**
+   * @return the user-defined categories which the annotated element belongs to
+   */
+  String[] value();
```

Comment:
🤔 This should be just a singular String, as the annotation is already repeatable.
I know that `@Issue` also has an array here, but that is because the repeated annotation support for extensions was added later and to keep backwards compatibility.

Reply:
At first, I just copied and pasted from `Issue` and wondered why it was both repeatable and accepted parameters. Later, I began to like it, because users have a choice of how they want to assign tags. Some like it more compact and write `@Tag(["A", "B", "C"])`, avoiding the more wordy and IDE screen real estate wasting

```java
@Tag("A")
@Tag("B")
@Tag("C")
```

The more groovy ones might use the compact form why the Java people might be more used to repeating annotations. I myself would prefer the compact form. Also, it would be inconsistent to have two similar annotation types with different features, and when somebody asks why, we have to reply: "The older one has more features because of historical reasons, but we deliberately decided that repeatable is enough, and therefore the more recent annotation has fewer features."

Reply:
> The more groovy ones might use the compact form why the Java people might be more used to repeating annotations.

I don't think this is java vs. groovy java also supports `@Tag({"A", "B", "C"})`.
The downside I see with having the list, is that you can't reliably search for `@Tag("B")`  and find all instances of it, when it could also be in a list with other tags.

> "The older one has more features because of historical reasons, but we deliberately decided that repeatable is enough, and therefore the more recent annotation has fewer features."

Having more variations to do the same things is not always a good thing.

Reply:
Yes, but here it is a good thing, looking at it from a user perspective. It avoids inconsistency with other extension tags and gives users ease of use and an easy choice.

From a Spock maintainer's perspective, it is both easy to implement and to maintain. There is no downside. The annotation visitors did all the work for me, it was absolutely trivial to implement them. The extension already transforms the grouped or repeated tags into single Spock model tag instances. Why would anyone want to iterate over the annotations from another place outside of an extension?

Reply:
As an additional data point, the JUnit team decided against making `Tag.value` and array to avoid the confusion of having two ways of declaring multiple tags. Arguably, one could expect `@Tags({"A", "B", "C"})` with an extra `s` but then going from a single tag to multiple tags would include a change in imports. In my experience, there's rarely a need to tag a specific test/feature method with multiple tags. Usually, in such cases, the class is tagged with one tag and the methods with another, additional one.

Reply:
Well, Spock is not Jupiter. Spock is about ease of use and concise syntax, somewhat in contrast to classical JUnit. I would still agree with your judgement, if other extensions would not have set a different precedence in Spock before. I think the confusion it could cause for users is not worth this "optimisation".